modules/datastore/src/Controller/QueryDownloadController.php

```diff
@@ -31,9 +26,8 @@ public function __construct(QueryService $queryService, DatasetInfo $datasetInfo
     // We do not want to cache streaming CSV content internally in Drupal,
     // because datasets can be very large. However, we do want CDNs to be able
     // to cache the CSV stream for a reasonable amount of time.
-    // @todo Replace this constant with some form of customizable caching
-    //   strategy.
-    $this->cacheMaxAge = static::RESPONSE_STREAM_MAX_AGE;
+    $config = $configFactory->get('datastore.settings');
+    $this->cacheMaxAge = $config->get('response_stream_max_age') ?: self::DEFAULT_RESPONSE_STREAM_MAX_AGE;
```

Comment:
The default value for the response stream max age is defined in `datastore.settings.yml`, so we don't need the constant here as a default value. The principle is that it's defined in only one place.

However, that also means we need an update hook to set the value in the site config.

Here's an example in the metastore module: https://github.com/GetDKAN/dkan/blob/2.x/modules/metastore/metastore.install#L113

Drupal docs here: https://www.drupal.org/docs/drupal-apis/update-api/updating-configuration#s-example-1-adding-a-configuration-key

Thanks!

Reply:
Added this to datastore.install and removed the constant.

/**
 * Set the default value for the response stream max age.
 */
function datastore_update_9002(&$sandbox) {
  $config = \Drupal::service('config.factory')->getEditable('datastore.settings');
  if ($config->get('response_stream_max_age') === NULL) {
    $config->set('response_stream_max_age', 3600)->save();
  }
}